ignite/handlers/running_average.py

```diff
@@ -0,0 +1,66 @@
+from ignite.metrics import Metric
+from ignite.engine import Events
+
+
+class RunningAverage(object):
+    """Compute running average of a metric or output
+
+    Args:
+        src (str or Metric): input source: "output" string or an instance of :class:`ignite.metrics.Metric`
+        alpha (float, optional): running average decay factor, default 0.98
+        output_transform (Callable, optional): a function to use to transform the output if `src` is "output".
+            Otherwise it should be None
+
+    """
+
+    def __init__(self, src, alpha=0.98, output_transform=None):
+        if not (isinstance(src, Metric) or src == 'output'):
+            raise TypeError("Argument src should be a Metric or string 'output'")
+        if not (0.0 < alpha <= 1.0):
+            raise ValueError("Argument alpha should be a float between 0.0 and 1.0")
+
+        if isinstance(src, Metric):
+            if output_transform is not None:
+                raise ValueError("Argument output_transform should be None if src is 'output'")
```

Comment:
This error message is confusing as this  is the case where `isinstance(src,Metric)` is `True`

Reply:
I think maybe we can just drop this entirely? I can see valid use cases where you way want to compute the running average of just some part of the output in which case we should allow `output_transform`.

Reply:
In case of `src` is `Metric` is it redundant to have `output_transform` as metric itself should take care of output transform. In case of src being engine.state.output it is better to be able to transform the output. This check somehow controls the redundant case in the way if user setups both `output_transform`s the code stops. But I'm OK to remove it

Reply:
I see you point about the redundancy in the `Metric` case. In that case, we can just update the error message, right? It currently reads "Argument output_transform should be None if src is 'output'" but should be "Argument output_transform should be None if src is a Metric"?

Reply:
Oh, yes, my bad. Error message is just wrong :)